### ISSUE TITLE
Attempt to retry saving of event in case database is locked

### ISSUE THREAD

**Issue by thomaseizinger** [itchysats/itchysats#1168]
We seem to be sometimes hitting the following error:

> error returned from database: database table is locked: EVENTS: database table is locked: EVENTS

https://github.com/itchysats/itchysats/runs/4876080983?check_suite_focus=true#step:6:266

It should be fairly straight-forward to retry saving the event in case of that error.

**Comment by da-kami** [comment]
I think this can happen relatively easily (here is another test run where it happened: https://github.com/itchysats/itchysats/runs/4890020429?check_suite_focus=true)

Should we tackle this before the next release?

**Comment by thomaseizinger** [comment]
Found this: https://github.com/launchbadge/sqlx/issues/1649

Temporary solutions seems to be to avoid concurrent connections on the pool level.

**Comment by thomaseizinger** [comment]
We should attempt doing #1211 again but see if #1218 is fixed if we manually close the connection after each use.

**Comment by luckysori** [comment]
I run into this compiler error, even if I `close` right after calling `acquire`:

```
error[E0507]: cannot move out of dereference of `sqlx::pool::PoolConnection<sqlx::Sqlite>`
  --> daemon/src/cfd_actors.rs:37:5
   |
37 |     conn.close().await;
   |     ^^^^^^^^^^^^ move occurs because value has type `sqlx::SqliteConnection`, which does not implement the `Copy` trait
```

Also, `sqlx`'s docs clearly state that the `close` method "is **not required** for safe and consistent operation".

**Comment by thomaseizinger** [comment]
> Also, `sqlx`'s docs clearly state that the `close` method "is **not required** for safe and consistent operation".

Interesting, this idea might be a wild-goose chase then!

**Comment by da-kami** [comment]
Given that the process manager is the only actor that writes to the database I wonder how we run into warnings like this one

```
WARN daemon::model::cfd: Rollover failed: Failed to process new domain event: error returned from database: database table is locked: EVENTS: database table is locked: EVENTS order_id=af55fa2b-e9c9-400b-b2da-c4c73dfb0ad3
```

This happened in a test run: https://github.com/itchysats/itchysats/runs/4963670054?check_suite_focus=true

How is this happening in this test? 
Are we trying to save the `accept` event so fast that the `start_rollover` one still locks the DB? What is going on here?

Maybe the database problem is actually not just a concurrency problem but we are using sqlx somewhat wrongly.

**Comment by rishflab** [comment]
https://github.com/launchbadge/sqlx/pull/1658/files/eff5f8ad43d69b4cad03830402f2f7592cc4ad47..140641875858c85d01ceb516a8f5415ca3f18a4b

Should be resolved with this ^ PR.

**Comment by thomaseizinger** [comment]
For reference, we fixed this in #1240 by cherry-picking the commits in https://github.com/launchbadge/sqlx/pull/1658 onto the `0.5.10` release of sqlx.
